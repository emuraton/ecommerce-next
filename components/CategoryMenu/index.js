import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  padding-left: 15px;
  padding-right: 15px;
  border: 1p
`;

const List = styled.ul`
  padding-left: 0px;
  list-style: none;
`;

const Item = styled.li`
  display: inline-block;
  margin-right: 20px;
`;

const CategoryMenu = ({ categories }) => {
  return (
    <Container>
      <List>
        {categories &&
          categories.map((category, index) => (
            <Item key={index}>
              <Category label={category} />
            </Item>
          ))}
      </List>
    </Container>
  );
};

const Category = ({ label }) => {
  return (
    <div>
      <Button>
        <Label>{label}</Label>
      </Button>
    </div>
  );
};

const Button = styled.button`
  position: relative;
  display: inline-block;
  list-style: none;
  padding: 12px 10px;
  border-radius: 25px;
  /* background-color: rgba(36, 37, 41, 0.05); */
  background-color: white;
  color: rgb(36, 37, 41);
  border-color: rgb(162, 164, 173);
  text-align: center;
  text-decoration: none;
  outline: none;
`;

const Label = styled.span`
  padding: 10px 15px;
`;

export default CategoryMenu;
