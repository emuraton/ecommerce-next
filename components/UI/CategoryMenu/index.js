import React from 'react';
import styled from 'styled-components';

import Category from '../Category';

const Container = styled.div`
  padding-left: 15px;
  padding-right: 15px;
`;

const List = styled.ul`
  padding-left: 0px;
  list-style: none;
`;

const Item = styled.li`
  display: inline-block;
  margin-right: 20px;
`;

const CategoryMenu = ({ categories, activeSlug, onClick }) => {
  return (
    <Container>
      <List>
        {categories &&
          categories.map(({ pk, name, slug }) => (
            <Item key={pk}>
              <Category
                name={name}
                slug={slug}
                active={slug === activeSlug}
                onClick={onClick}
              />
            </Item>
          ))}
      </List>
    </Container>
  );
};

export default CategoryMenu;
