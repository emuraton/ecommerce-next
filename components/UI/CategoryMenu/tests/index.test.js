import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CategoryMenu from '../';

describe('<CategoryMenu />', () => {
  const initialProps = {
    categories: [
      { pk: 1, slug: 'ticket', name: 'Ticket', products: [] },
      { pk: 2, slug: 'hotel', name: 'Hotel', products: [] },
    ],
    slideIndex: 1,
    onClick: Function.prototype,
  };
  describe('@renders', () => {
    it('default render', () => {
      const wrapper = shallow(<CategoryMenu {...initialProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('no categories, no display', () => {
      const props = { ...initialProps, categories: null };
      const wrapper = shallow(<CategoryMenu {...props} />);
      expect(toJson(wrapper)).toBe('');
    });
  });
});
