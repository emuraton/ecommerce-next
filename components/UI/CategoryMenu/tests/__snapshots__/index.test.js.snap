// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryMenu /> @renders default render 1`] = `
<CategoryMenu__Container>
  <CategoryMenu__List>
    <CategoryMenu__Item
      key="1"
    >
      <Category
        active={false}
        name="Ticket"
        slug="ticket"
      />
    </CategoryMenu__Item>
    <CategoryMenu__Item
      key="2"
    >
      <Category
        active={false}
        name="Hotel"
        slug="hotel"
      />
    </CategoryMenu__Item>
  </CategoryMenu__List>
</CategoryMenu__Container>
`;
